flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/debezium/DebeziumJsonSerializationSchema.java

```diff
@@ -0,0 +1,112 @@
+import org.apache.flink.formats.json.JsonRowDataSerializationSchema;
+import org.apache.flink.formats.json.TimestampFormat;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.types.RowKind;
+
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static org.apache.flink.table.types.utils.TypeConversions.fromLogicalToDataType;
+
+/**
+ * Serialization schema from Flink Table/SQL internal data structure {@link RowData} to Debezium JSON.
+ *
+ * @see <a href="https://debezium.io/">Debezium</a>
+ */
+public class DebeziumJsonSerializationSchema implements SerializationSchema<RowData> {
+	private static final long serialVersionUID = 1L;
+
+	private static final StringData OP_INSERT = StringData.fromString("INSERT"); // insert
+	private static final StringData OP_DELETE = StringData.fromString("DELETE"); // delete
+
+	/** The serializer to serialize Debezium JSON data. **/
+	private final JsonRowDataSerializationSchema jsonSerializer;
+
+	private transient GenericRowData genericRowData;
+
+	public DebeziumJsonSerializationSchema(RowType rowType, TimestampFormat timestampFormat) {
+		jsonSerializer = new JsonRowDataSerializationSchema(
+			createJsonRowType(fromLogicalToDataType(rowType)),
+			timestampFormat);
+	}
+
+	@Override
+	public void open(InitializationContext context) {
+		genericRowData = new GenericRowData(2);
+	}
+
+	@Override
+	public byte[] serialize(RowData rowData) {
+		try {
+			genericRowData.setField(0, rowData);
+			genericRowData.setField(1, rowKind2String(rowData.getRowKind()));
+			return jsonSerializer.serialize(genericRowData);
+		} catch (Throwable t) {
+			throw new RuntimeException(format("Could not serialize row '%s'.", rowData), t);
+		}
+	}
+
+	private StringData rowKind2String(RowKind rowKind) {
+		switch (rowKind) {
+			case INSERT:
+			case UPDATE_AFTER:
+				return OP_INSERT;
+			case UPDATE_BEFORE:
+			case DELETE:
+				return OP_DELETE;
+			default:
+				throw new UnsupportedOperationException(format("Unsupported operation '%s' for row kind.", rowKind));
+		}
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+		DebeziumJsonSerializationSchema that = (DebeziumJsonSerializationSchema) o;
+		return Objects.equals(jsonSerializer, that.jsonSerializer);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(jsonSerializer);
+	}
+
+	private static RowType createJsonRowType(DataType databaseSchema) {
+		// Debezium JSON contains some other information, e.g. "source", "ts_ms"
+		// but we don't need them.
+		return (RowType) DataTypes.ROW(
+			DataTypes.FIELD("after", databaseSchema),
```

Comment:
This is wrong. Debezium serialize insert data in "after", but delete data in "before".